### PR TITLE
Retarget DOTNETHOME when installing x64 on ARM64

### PR DESCRIPTION
**Pull request by ericstj**
Replaces https://github.com/dotnet/runtime/pull/58059

### CODE REVIEW

**Comment by am11** [comment]
Could it be renamed to something like `DOTNET_BASEDIR`? Otherwise, this `DOTNETHOME` (directory name) and the existing `DOTNET_HOME` (well-documented environment variable pointing to base dir) can cause ambiguity in certain contexts.

**Comment by ericstj** [comment]
> Could it be renamed to something like DOTNET_BASEDIR

This is not a new identifier.  This a pre-existing directory identifier and public property exposed by the MSI installers.  Changing it is breaking.  All these changes do is detect when x64 is being installed on ARM64, and the user hasn't specified DOTNETHOME, redirect to an x64 subfolder.  See https://github.com/dotnet/arcade/pull/7785

**Comment by ericstj** [comment]
Ok reviewers, this should be ready to review now.  https://github.com/dotnet/runtime/pull/58751 brought the arcade changes over that enable this.  I'd like to get this in ASAP so we have 7.0 builds to smoke test on ARM64, then we can request the port for this to 6.0.  Thanks!

File: src/installer/pkg/sfx/installers/host.wxs [hunk 1]
```diff
@@ -47,6 +62,16 @@
...
+    <?if $(var.Platform)~=x64 ?>
+    <CustomActionRef Id="Set_PROGRAMFILES_DOTNET_NON_NATIVE_ARCHITECTURE" />
+    <?endif?>
   </Fragment>
-
-</Wix>
+  
+  <Fragment>
+    <!-- Unlike DOTNETHOME which gives precedence to a user specified value over an x64 suffix, here we always want the suffixed path -->
+    <SetProperty Action="Set_PROGRAMFILES_DOTNET_NON_NATIVE_ARCHITECTURE" Id="PROGRAMFILES_DOTNET" Value="[$(var.Program_Files)]dotnet\x64\" After="Set_NON_NATIVE_ARCHITECTURE">
```
Comment (NikolaMilosavljevic) [inline src/installer/pkg/sfx/installers/host.wxs:73]: Is this action defined in a separate file?

Comment (ericstj) [reply src/installer/pkg/sfx/installers/host.wxs:73]: Yes https://github.com/dotnet/arcade/pull/7785/files#diff-0938a95b0962aaae248064ee37481237c283dac60ea91ebc3ef38138bbbb52fbR23

**Review by NikolaMilosavljevic — APPROVED** [review]

**Review by hoyosjs — COMMENTED** [review]
LGTM

**Pull request merged**